target/linux/ipq40xx/files/arch/arm/boot/dts/qcom-ipq4018-wap-ac.dts

```diff
@@ -187,6 +187,24 @@
 	status = "okay";
 };
 
+&gmac {
+	status = "okay";
+};
+
+&switch {
+	status = "okay";
+};
+
+&swport4 {
+	status = "okay";
+	label = "sw-eth2";
```

Comment:
Why can't it be labeled eth1 and eth2 in the same way as on the case?

Reply:
That’s what I had [originally done](https://github.com/markmentovai/openwrt/commit/98a344b5b206efa26c235f1cd334cf5354c748cf#diff-092403a5868cf4896adf73ced6d5de05cf805084f47a77ab0280dc98f397e1acR199) (and [an explanation](https://github.com/openwrt/openwrt/pull/4721#issuecomment-1002229806)), but [doing using `eth` for DSA switch ports was discouraged](https://github.com/openwrt/openwrt/pull/4721#issuecomment-1020135423) as they’re not true GMACs. I can see how having `eth0` and (`eth1`, `eth2`) behave differently would be quite confusing. [I explained the `sw-` prefix](https://github.com/openwrt/openwrt/pull/4721#issuecomment-1046103519) in more detail later in that thread.

Reply:
Some would prefer to avoid `eth%d` labels because these are the Linux default network interface names, and we do not want to collide with them (on other platforms especially).
Could call them `ether1`, `ether2`. I think this is what stock (RouterOS) does?

Reply:
There isn’t really any precedent for naming interfaces according to the stock OS, and I think `ether1` is too similar to `eth1`.

[Explanation of the `sw-` naming](https://github.com/openwrt/openwrt/pull/4721#issuecomment-1046103519), but summarizing:

 - There’s a preference to name ports based on how they’re labeled on the case, for clarity and user convenience. The ports are labeled ETH1 and ETH2 on the case, but `eth1` and `eth2` are not good choices because they’re in the `eth` namespace. (There’s still an `eth0` for the true GMAC.)
 - There’s nothing on the case that indicates function like LAN or WAN, so choosing `lan` or `wan` names is not appropriate.
 - The OpenWrt default configuration bridges both ports into the LAN network, but `lan1` and `lan2` are still not appropriate. If the user configuration changes the function of the ports, such as using ETH1 for `lan` and ETH2 for `wan`, it would be incredibly confusing to have network `wan` associated with port `lan2`, when LAN isn’t even the labeled name of the port.
 - The `sw-` prefix makes clear that they’re switch ports.
 - `sw-` follows the `br-` precedent that OpenWrt uses for bridge interfaces (like the common `br-lan`).
 - The interface names remain true to the labels printed on the case (ETH1 and ETH2) but with a prefix (`sw-`). Descriptive, concise, and minimally confusing.